data/lib/core/creature.lua

```diff
@@ -1,29 +1,25 @@
-function Creature.getClosestFreePosition(self, position, extended)
-	local usePosition = Position(position)
-	local tiles = { Tile(usePosition) }
-	local length = extended and 2 or 1
+function Creature.getClosestFreePosition(self, position, maxRadius, mustBeReachable)
+	maxRadius = maxRadius or 1
 
-	local tile
-	for y = -length, length do
-		for x = -length, length do
-			if x ~= 0 or y ~= 0 then
-				usePosition.x = position.x + x
-				usePosition.y = position.y + y
+	-- backward compatability (extended)
+	if maxRadius == true then
+		maxRadius = 2
+	end
 
-				tile = Tile(usePosition)
-				if tile then
-					tiles[#tiles + 1] = tile
+	for radius = 0, maxRadius do
+		for x = -radius, radius do
+			for y = -radius, radius do
+				if math.abs(x) == radius or math.abs(y) == radius then
```

Comment:
Yeah, it's pretty hard to get a perfect spiraling loop :P a good start would be instead of looping from -radius to radius, using only `radius` and duplicating code (not a big issue considering reduced complexity).